### PR TITLE
opengl: fix Qt warnings

### PR DESCRIPTION
**Pull request by oltolm**
I tried fixing the Qt warnings that are printed when you compile RPCS3 in debug mode.

```
warning: QOpenGLContext::swapBuffers() called with non-exposed window, behavior is undefined
```
```
warning: Attempting to create QWindow-based QOffscreenSurface outside the gui thread. Expect failures.
```

### CODE REVIEW

File: rpcs3/rpcs3qt/gl_gs_frame.cpp [hunk 1]
```diff
@@ -29,13 +31,16 @@ draw_context_t gl_gs_frame::make_context()
 
 	if (m_primary_context)
 	{
-		auto surface = new QOffscreenSurface();
-		surface->setFormat(m_format);
-		surface->create();
+		atomic_t<QOffscreenSurface*> surface = new QOffscreenSurface();
```
Comment (elad335) [inline rpcs3/rpcs3qt/gl_gs_frame.cpp:34]: It doesn't have to be atomic, but it doesn't hurt I guess.

Comment (oltolm) [reply rpcs3/rpcs3qt/gl_gs_frame.cpp:34]: Interesting. How do you synchronize access without atomic in C++?

Comment (elad335) [reply rpcs3/rpcs3qt/gl_gs_frame.cpp:34]: This is not cuncurrent access, cuncurrency is what std::atomic (or atomic_t here) is for.

Comment (oltolm) [reply rpcs3/rpcs3qt/gl_gs_frame.cpp:34]: OK, I thought that `surface` is used in two threads: the one calling `make_context` and the main thread. I guess I don't understand it correctly.

Comment (elad335) [reply rpcs3/rpcs3qt/gl_gs_frame.cpp:34]: It is used by 2 threads, but cannot at the same time. That is the difference.

Comment (Megamouse) [reply rpcs3/rpcs3qt/gl_gs_frame.cpp:34]: It doesn't make sense to create an atomic here if it's not used anywhere else

Comment (oltolm) [reply rpcs3/rpcs3qt/gl_gs_frame.cpp]: I removed it.

**Review by elad335 — APPROVED** [review]

File: rpcs3/rpcs3qt/gl_gs_frame.cpp [hunk 2]
```diff
@@ -29,9 +30,12 @@ draw_context_t gl_gs_frame::make_context()
 
 	if (m_primary_context)
 	{
-		auto surface = new QOffscreenSurface();
+		QOffscreenSurface* surface = new QOffscreenSurface();
```
Comment (kd-11) [inline rpcs3/rpcs3qt/gl_gs_frame.cpp:33]: as opposed to what? this is redundant

File: rpcs3/rpcs3qt/gl_gs_frame.cpp [hunk 3]
```diff
@@ -29,9 +30,12 @@ draw_context_t gl_gs_frame::make_context()
...
 		surface->setFormat(m_format);
-		surface->create();
+		Emu.BlockingCallFromMainThread([&]()
+		{
+			surface->create();
```
Comment (kd-11) [inline rpcs3/rpcs3qt/gl_gs_frame.cpp:37]: Why? You can hand over opengl contexts between threads using glMakeCurrent

Comment (kd-11) [reply rpcs3/rpcs3qt/gl_gs_frame.cpp:37]: In fact this is why the set_current method exists. The thread calling set_current "acquires" the gl render context. Such hacks aren't necessary.

File: rpcs3/rpcs3qt/gl_gs_frame.cpp (hunk 2 above)
Comment (oltolm) [reply rpcs3/rpcs3qt/gl_gs_frame.cpp]: It was an oversight. Reverted.

File: rpcs3/rpcs3qt/gl_gs_frame.cpp (hunk 3 above)
Comment (oltolm) [reply rpcs3/rpcs3qt/gl_gs_frame.cpp:37]: This [note](https://doc.qt.io/qt-5/qoffscreensurface.html#details) is from the Qt documentation.

> Note: Due to the fact that `QOffscreenSurface` is backed by a [`QWindow`](https://doc.qt.io/qt-5/qwindow.html) on some platforms, cross-platform applications must ensure that [`create`](https://doc.qt.io/qt-5/qoffscreensurface.html#create)() is only called on the main (GUI) thread. The `QOffscreenSurface` is then safe to be used with [`makeCurrent()`](https://doc.qt.io/qt-5/qopenglcontext.html#makeCurrent) on other threads, but the initialization and destruction must always happen on the main (GUI) thread.

Comment (kd-11) [reply rpcs3/rpcs3qt/gl_gs_frame.cpp:37]: Just the kind of hacky BS one should expect from Qt xD

Comment (kd-11) [reply rpcs3/rpcs3qt/gl_gs_frame.cpp:37]: Add a comment explaining that this anti-pattern is a workaround to a Qt issue. That way it doesn't get removed in future.

Comment (oltolm) [reply rpcs3/rpcs3qt/gl_gs_frame.cpp:38]: I added a comment.

**Pull request merged**